content/800-guides/1200-prisma-postgres-realtime-on-cloudflare.mdx

```diff
@@ -0,0 +1,451 @@
+    return c.text("Invalid input: x and y must be numbers.", 400);
+  }
+
+  const prisma = createPrismaClient(c.env.DATABASE_URL, c.env.PULSE_API_KEY);
+
+  const newPoint = await prisma.points.create({ data: { x, y } });
+  return c.json({ point: newPoint });
+});
+```
+
+### 6. Create a `GET` route that serves an HTML page  
+
+This route serves an HTML page with a form and scatter plot. Also establishes a connection to the websocket route and receives and reflects events from Pulse in real-time.
+
+
+```typescript file=src/index.ts
+app.get("/", async (c) => {
+  const prisma = createPrismaClient(c.env.DATABASE_URL, c.env.PULSE_API_KEY);
+  const dataPoints = await prisma.points.findMany({
+    take: 100,
+    orderBy: { id: "desc" },
+    select: { x: true, y: true },
+  });
+
+  return c.html(`
+    <!DOCTYPE html>
+    <html>
+    <head>
+      <title>Real-Time Scatter Plot</title>
+      <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>
+    </head>
+    <body>
+      <form id="addPointForm">
+        <input name="x" type="number" placeholder="X coordinate" required />
+        <input name="y" type="number" placeholder="Y coordinate" required />
+        <button type="submit">Submit</button>
+      </form>
+
+      <canvas id="scatterPlot"></canvas>
+
+      <script>
+        const dataPoints = ${JSON.stringify(dataPoints)};
+        const ctx = document.getElementById('scatterPlot').getContext('2d');
+        const chart = new Chart(ctx, {
+          type: 'scatter',
+          data: { datasets: [{ data: dataPoints }] },
+        });
+
+        document.getElementById('addPointForm').onsubmit = async (e) => {
+          e.preventDefault();
+          const form = new FormData(e.target);
+          await fetch("/", {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({ x: +form.get("x"), y: +form.get("y") }),
+          });
+          e.target.reset();
+        };
+
+        const ws = new WebSocket("ws://" + window.location.host + "/ws");
+        ws.onmessage = (event) => {
+          const newPoint = JSON.parse(event.data);
+          chart.data.datasets[0].data.push(newPoint);
+          chart.update();
+        };
+      </script>
+    </body>
+    </html>
+  `);
+});
+
+export default app;
+```
+
+### 7. Start the server and test your application  
+
+Run the development server:  
+
+```terminal
+npm run dev
+```
+
+Visit `https://localhost:8787` to see your app in action. You should see a form where, if you input data, the scatter plot updates in real time:
+
+![An demo of the app we built where a scatter plot updates in real-time](/img/guides/realtime-app-demo-ppg-cf-workers.gif)
+
+## 5. Deploy the application to Cloudflare
+
+Now you'll deploy your real-time application to Cloudflare Workers. This involves uploading your application code and securely configuring your environment variables.
+
+### 1. Set up a Cloudflare account
```

Comment:
since you mentioned this as a prerequisite, I don't think you need to include this step.